driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -213,9 +215,10 @@ public static CodecRegistry getDefaultCodecRegistry() {
 
     private static final CodecRegistry DEFAULT_CODEC_REGISTRY =
             fromProviders(asList(new ValueCodecProvider(),
-                    new DocumentCodecProvider(),
                     new BsonValueCodecProvider(),
-                    new IterableCodecProvider(),
+                    new DBRefCodecProvider(),
+                    new DocumentCodecProvider(new DocumentToDBRefTransformer()),
```

Comment:
I'd rather bring the sync and async codecs into line. Potentially, any changes to the default codec will produce backwards breaking changes to user land code. I'd rather that we allow changes to the default. That said we've only had one person ask for DBRef support in async so it probably isn't that popular inside the driver.